src/sip/auth.c

```diff
@@ -155,6 +193,10 @@ static bool auth_handler(const struct sip_hdr *hdr, const struct sip_msg *msg,
 		if (err)
 			goto out;
 
+		err = pl_strdup(&realm->algorithm, &ch.algorithm);
```

Comment:
I have added a new test case https://github.com/baresip/retest/commit/78ea4df1d471b1b4b23e1705433b950bcf371071
If algorithm is not specified, pl_strdup fails with EINVAL.